MitoMathHelper.js

```diff
@@ -4,113 +4,158 @@
-            circleB.getPosition(),
+        let relativePositionA = circleA.getPosition();
+        let relativePositionB = circleB.getPosition();
+        let positionA = [positionOffsetA[0] + relativePositionA[0], positionOffsetA[1] + relativePositionA[1]];
+        let positionB = [positionOffsetB[0] + relativePositionB[0], positionOffsetB[1] + relativePositionB[1]];
+        let radiusA = circleA.getRadius();
+        let radiusB = circleB.getRadius();
+
+        let distanceA = [velocityA[0] * interval, velocityA[1] * interval];
+        let invertedDistanceB = [-velocityA[0] * interval, -velocityA[1] * interval];
+        let combinedDistance = MitoMathHelper._addPoints(distanceA, invertedDistanceB);
+        let combinedMagnitude = Math.hypot(combinedDistance[0], combinedDistance[1]);
+
+        // Calculate closest point on combined velocities to circleB
+        let positionACombinedDistancePoint = [
+            positionA[0] + combinedDistance[0],
+            positionA[1] + combinedDistance[1]
+        ];
+        let positionACombinedDistanceClosestPoint = MitoMathHelper._getClosestPointOnLine(
+            [positionA, positionACombinedDistancePoint],
+            positionB,
         );
 
-        //If closest circle is in range calculate back off distance
-        //Tick percentage should be between 0 - 1
-        let distance = MitoMathHelper._distanceBetweenTwoPointsSquared(closestPoint, circleB.getPosition());
-        let radiusTotal = circleA.getRadius() + circleB.getRadius();
-        radiusTotal *= radiusTotal;
-        if (radiusTotal >= distance) {
-            let backOff = Math.sqrt(radiusTotal - distance);
-            let backPoint = [backOff * (combineVelocity[0] / combineMagnitude), backOff * (combineVelocity[1] / combineMagnitude)];
-            let closestPointWithinVelocity = MitoMathHelper._addPoints(closestPoint, MitoMathHelper._invertPoint(backPoint));
-            let distanceBetween = MitoMathHelper._distanceBetweenTwoPoints(circleA.getPosition(), closestPointWithinVelocity);
-            let tickPercentage = distanceBetween / combineMagnitude;
-            if (tickPercentage < 0 || tickPercentage > 1) {
-                return null;
-            }
-
-            return tickPercentage;
+        // If closest circle is in range calculate back off distance
+        // Tick percentage should be between 0 - 1
+        let positionBToClosestPointDistanceSquared = MitoMathHelper._distanceSquaredBetweenTwoPoints(positionACombinedDistanceClosestPoint, positionB);
+        let radiusTotal = radiusA + radiusB;
+        let radiusTotalSquared = radiusTotal * radiusTotal;
+        if (radiusTotalSquared > positionBToClosestPointDistanceSquared) {
+            return null
         }
 
-        return null;
+        let backOffAmount = Math.sqrt(radiusTotalSquared - positionBToClosestPointDistanceSquared); // TODO shouldn't this be Math.sqrt(radiusTotalSquared) - Math.sqrt(positionBToClosestPointDistanceSquared) ?????
```

Comment:
No this is calculating one side of the triangle sqrt(dx^2 + dy^2) = back off distance